### PR TITLE
fix(static): safe decode path

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/22071

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

In some cases (such as a malformed URL) we might encounter an error when checking whether the URL is a static asset. However, we can be assured it is _not_ a static asset and either throw a 404 if it has the static prefix, or simply early return from the handler and directly progress to the next handler, which can allow the user to correct the URL or throw their own error.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
I think we can use `ufo.decodePath` instead which has built-in try/catch safety (https://github.com/unjs/ufo/blob/main/src/encoding.ts#L124)

File: src/runtime/static.ts [hunk 1]
```diff
@@ -15,38 +15,41 @@ export default eventHandler((event) => {
...
+        const _asset = getAsset(_id);
+        if (_asset) {
+          asset = _asset;
+          id = _id;
+          break;
+        }
       }
     }
-  }
+  } catch {}
```
Comment (pi0) [inline src/runtime/static.ts:52]: Why we need a try catch here with wide scope?

Comment (danielroe) [reply src/runtime/static.ts:52]: It was so we could avoid further normalising of the id if it couldn't be resolved. I'll push a refactor with `decodePath`.

Comment (pi0) [reply src/runtime/static.ts:52]: Thanks. Would be nice if we could limit the scope of try/catch this way to avoid hiding other errors.

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**